infrastructure/builder/src/generate/generators/k8s.js

```diff
@@ -0,0 +1,354 @@
+const _ = require('lodash');
+const path = require('path');
+const glob = require('glob');
+const util = require('util');
+const yaml = require('js-yaml');
+const jsone = require('json-e');
+const config = require('taskcluster-lib-config');
+const rimraf = util.promisify(require('rimraf'));
+const mkdirp = util.promisify(require('mkdirp'));
+const {listServices, writeRepoFile, readRepoYAML, writeRepoYAML, writeRepoJSON, REPO_ROOT, configToSchema, configToExample} = require('../../utils');
+
+const SERVICES = listServices();
+const CHART_DIR = path.join('infrastructure', 'k8s');
+const TMPL_DIR = path.join(CHART_DIR, 'templates');
+
+const CLUSTER_DEFAULTS = {
+  level: 'notice',
+  node_env: 'production',
+
+  // TODO: iirc google doesn't set the headers that we need to trust proxy so we don't set this, let's fix it
+  force_ssl: false,
+  trust_proxy: true,
+};
+
+const NON_CONFIGURABLE = [
+  'port',
+  'taskcluster_root_url', // This is actually configured at the cluster level
+];
+
+const renderTemplates = async (name, vars, procs, templates) => {
+
+  await rimraf(TMPL_DIR);
+  await mkdirp(TMPL_DIR);
+
+  for (const resource of ['role', 'rolebinding', 'serviceaccount', 'secret']) {
+    const rendered = jsone(templates[resource], {
+      projectName: `taskcluster-${name}`,
+      secrets: vars.map(v => ({
+        key: v,
+        val: `.Values.${name.replace(/-/g, '_')}.${v.toLowerCase()}`,
+      })),
+    });
+    const file = `taskcluster-${name}-${resource}.yaml`;
+    await writeRepoYAML(path.join(TMPL_DIR, file), rendered);
+  }
+
+  const ingresses = [];
+  for (const [proc, conf] of Object.entries(procs)) {
+    let tmpl;
+    const context = {
+      projectName: `taskcluster-${name}`,
+      serviceName: name,
+      configName: name.replace(/-/g, '_'),
+      configProcName: proc.replace(/-/g, '_'),
+      procName: proc,
+      needsService: false,
+      readinessPath: conf.readinessPath || `/api/${name}/v1/ping`,
+    };
+    switch (conf['type']) {
+      case 'web': {
+        tmpl = 'deployment';
+        context['needsService'] = true;
+        const rendered = jsone(templates['service'], context);
+        const file = `taskcluster-${name}-service-${proc}.yaml`;
+        ingresses.push({
+          projectName: `taskcluster-${name}`,
+          paths: conf['paths'] || [`/api/${name}/*`], // TODO: This version of config is only for gcp ingress :(
```

Comment:
Could Helm conditionals take care of this?

Reply:
Yes! Created bug 1566928 for followup